subsys/net/l2/wifi/wifi_utils.c

```diff
@@ -223,13 +225,21 @@ int wifi_utils_parse_scan_bands(char *scan_bands_str, uint8_t *band_map)
 		return -EINVAL;
 	}
 
-	band_str = strtok_r(scan_bands_str, ",", &ctx);
+	parse_str = strdup(scan_bands_str);
+
+	if (!parse_str) {
+		NET_ERR("Unable to duplicate scan bands string");
```

Comment:
`duplicate` would just confuse the user, we should use `Failed to allocate memory: %d bytes, err: %s`